lib/archethic_web/live/transaction_details_live.ex

```diff
@@ -78,32 +83,86 @@ defmodule ArchethicWeb.TransactionDetailsLive do
 
   defp handle_transaction(
          socket,
-         tx = %Transaction{address: address}
+         tx = %Transaction{
+           address: address,
+           data: %TransactionData{
+             ledger: %Ledger{token: %TokenLedger{transfers: token_transfers}}
+           },
+           validation_stamp: %ValidationStamp{
+             ledger_operations: %LedgerOperations{transaction_movements: transaction_movements}
+           }
+         }
        ) do
     previous_address = Transaction.previous_address(tx)
 
-    with {:ok, balance} <- Archethic.get_balance(address),
-         inputs <- Archethic.get_transaction_inputs(address) do
-      ledger_inputs = Enum.reject(inputs, &(&1.type == :call))
-      contract_inputs = Enum.filter(inputs, &(&1.type == :call))
-      uco_price_at_time = tx.validation_stamp.timestamp |> OracleChain.get_uco_price()
-      uco_price_now = DateTime.utc_now() |> OracleChain.get_uco_price()
+    inputs = Archethic.get_transaction_inputs(address)
+    ledger_inputs = Enum.reject(inputs, &(&1.type == :call))
+    contract_inputs = Enum.filter(inputs, &(&1.type == :call))
+    uco_price_at_time = tx.validation_stamp.timestamp |> OracleChain.get_uco_price()
+    uco_price_now = DateTime.utc_now() |> OracleChain.get_uco_price()
+
+    tokens =
```

Comment:
If you want to reduce lines and avoid repetition you could use recursive function:

```elixir

token_properties = []
 |> get_token_addresses(ledger_inputs)
 |> get_token_addresses(transaction_movements)
 |> get_token_addresses(token_transfers)
 |> Enum.uniq()
 |> get_token_properties()

# ...

defp get_token_addresses(acc, [ %TransactionMovement{ type: {:token, token_address, _}} | rest. ]) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses(acc, [ %TransactionInput{ type: {:token, token_address, _}} | rest. ]) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses([ %TokenTransfer{ token_address: token_address } | rest. ], acc) do
    get_token_addresses([ token_address | acc ], rest)
end

defp get_token_addresses(acc, _) do
    get_token_addresses(acc, rest)
end


```